src/XeroPHP/Application.php

```diff
@@ -108,6 +104,48 @@ public function getConfig($key)
         return $this->config[$key];
     }
 
+    /**
+    * @param string $config
+    * @param mixed $option
+    * @param mixed $value
+    * @return mixed
+    * @throws Exception
+    */
+    public function getConfigOption($config, $option) {
+        if (!isset($this->config[$config])) {
+            throw new Exception("Invalid configuration key [$key]");
+        }
+        return $this->config[$config][$option];
+    }
+
+    /**
+     * @param array $config
+     * @return array
+     */
+    public function setConfig($config) {
+        $this->config = array_replace_recursive(
+            self::$_config_defaults,
+            static::$_type_config_defaults,
+            $config
+        );
+
+        return $this->config;
+    }
+
+    /**
+     * @param string $config
+     * @param mixed $option
+     * @param mixed $value
+     * @return array
+     * @throws Exception
+     */
+    public function setConfigOption($config, $option, $value) {
+        if (!isset($this->config[$config])) {
+            throw new Exception("Invalid configuration key [$key]");
+        }
+        $this->config[$config][$option] = $value;
```

Comment:
Thanks for reviewing.
It assumes it will be like the $_config_defaults array, the first key is the group name (e.g, xero) and the second key is the option name. E.g., I would be calling setConfigOption('curl', 'CURLOPT_USERAGENT', 'Josh-G'); to set the useragent. If you'd like the behaviour changed at all, please let me know and I will amend the pull req